t/Hydra/Config/ldap_role_map.t

```diff
@@ -0,0 +1,242 @@
+
+use strict;
+use warnings;
+use Setup;
+use Hydra::Config;
+use Test2::V0;
+
+my $tmpdir = File::Temp->newdir();
+my $cfgfile = "$tmpdir/conf";
+my $scratchCfgFile = "$tmpdir/hydra.scratch.conf";
+
+my $ldapInHydraConfFile = "$tmpdir/hydra.empty.conf";
+write_file($ldapInHydraConfFile, <<CONF);
+<ldap>
+    <config>
+        <credential>
+            class = Password
+        </credential>
+    </config>
+    <role_mapping>
+        hydra_admin = admin
+        hydra_one_group_many_roles = create-projects
+        hydra_one_group_many_roles = cancel-build
+    </role_mapping>
+</ldap>
+CONF
+my $ldapInHydraConf = Hydra::Config::loadConfig($ldapInHydraConfFile);
+
+my $emptyHydraConfFile = "$tmpdir/hydra.empty.conf";
+write_file($emptyHydraConfFile, "");
+my $emptyHydraConf = Hydra::Config::loadConfig($emptyHydraConfFile);
+
+my $ldapYamlFile = "$tmpdir/ldap.yaml";
+write_file($ldapYamlFile, <<YAML);
+credential:
+    class: Password
+YAML
+
+subtest "getLDAPConfig" => sub {
+     subtest "No ldap section and an env var gets us legacy data" => sub {
```

Comment:
A nit, but some of the spacing in this sub looks a bit wonky, e.g. this first line is indented by 5 spaces.

Reply:
D'oh.